lib/Service/Social/InstagramProvider.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Service\Social;
+
+use OCP\Http\Client\IClientService;
+
+class InstagramProvider implements ISocialProvider {
+
+	/** @var IClientService */
+	private $httpClient;
+
+	public function __construct(IClientService $httpClient) {
+		$this->httpClient = $httpClient->NewClient();
+	}
+	
+	/**
+	 * Returns the profile-id
+	 *
+	 * @param {string} the value from the contact's x-socialprofile
+	 *
+	 * @return string
+	 */
+	public function cleanupId(string $candidate):?string {
```

Comment:
@call-me-matt on those functions, please directly use the strict type hint.
`basename` always returns a string :)

```suggestion
	public function cleanupId(string $candidate): string {
```

Reply:
Same goes for all the other social providers files :)

Reply:
For Instagram that is true, but Mastodon or Twitter providers, for example, could return NULL. Thought it would be easier to handle upcoming errors this way. Do you think it's not a good idea?

Reply:
The goal to have strict typing s to make sure we know what to expect :)
If there is indeed a potential null, let's leave it like that for the Mastodon or Twitter provider :rocket: